crates/typst-library/src/compute/data.rs

```diff
@@ -296,48 +369,53 @@ pub fn toml(
     let id = vm.location().join(&path).at(span)?;
     let data = vm.world().file(id).at(span)?;
 
+    toml_decode(Spanned::new(Readable::Bytes(data), span))
+}
+
+/// Reads structured data from a TOML string/bytes.
+///
+/// The string/bytes must contain a valid TOML table. TOML tables will be
+/// converted into Typst dictionaries, and TOML arrays will be converted into
+/// Typst arrays. Strings, booleans and datetimes will be converted into the Typst
+/// equivalents and numbers will be converted to floats or integers depending on
+/// whether they are whole numbers.
+///
+/// Display: TOML
+/// Category: data-loading
+#[func]
+pub fn toml_decode(
+    /// TOML data.
+    data: Spanned<Readable>,
+) -> SourceResult<Value> {
+    let Spanned { v: data, span } = data;
+    let data: Bytes = data.into();
     let raw = std::str::from_utf8(&data)
         .map_err(|_| "file is not valid utf-8")
         .at(span)?;
 
-    let value: toml::Value = toml::from_str(raw).map_err(format_toml_error).at(span)?;
-    Ok(convert_toml(value))
+    let value: Value = toml::from_str(raw).map_err(format_toml_error).at(span)?;
+    Ok(value)
 }
 
-/// Convert a TOML value to a Typst value.
-fn convert_toml(value: toml::Value) -> Value {
-    match value {
-        toml::Value::String(v) => v.into_value(),
-        toml::Value::Integer(v) => v.into_value(),
-        toml::Value::Float(v) => v.into_value(),
-        toml::Value::Boolean(v) => v.into_value(),
-        toml::Value::Array(v) => {
-            v.into_iter().map(convert_toml).collect::<Array>().into_value()
-        }
-        toml::Value::Table(v) => v
-            .into_iter()
-            .map(|(key, value)| (key.into(), convert_toml(value)))
-            .collect::<Dict>()
-            .into_value(),
-        toml::Value::Datetime(v) => match (v.date, v.time) {
-            (None, None) => Value::None,
-            (Some(date), None) => {
-                Datetime::from_ymd(date.year as i32, date.month, date.day).into_value()
-            }
-            (None, Some(time)) => {
-                Datetime::from_hms(time.hour, time.minute, time.second).into_value()
-            }
-            (Some(date), Some(time)) => Datetime::from_ymd_hms(
-                date.year as i32,
-                date.month,
-                date.day,
-                time.hour,
-                time.minute,
-                time.second,
-            )
-            .into_value(),
-        },
-    }
+/// Encode structured data into a toml string.
+///
+/// Display: TOML
+/// Category: data-loading
+#[func]
+pub fn toml_encode(
+    /// Value to be encoded
+    value: Spanned<Value>,
+    /// Pretty print
```

Comment:
should be a bit more explanation here, too. I don't know exactly what non-pretty-printed toml would like actually

Reply:
The docs just say: "Apply a default “pretty” policy to the document"

Reply:
You don't have to write much, the comment should just be a bit more friendly than `Pretty print`. And all comments should end with a period.